sync/sync.go

```diff
@@ -0,0 +1,345 @@
+package sync
+
+import (
+	"fmt"
+	"github.com/gogo/protobuf/proto"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/mesh"
+	"github.com/spacemeshos/go-spacemesh/p2p"
+	"github.com/spacemeshos/go-spacemesh/sync/pb"
+	"sync/atomic"
+	"time"
+)
+
+type Block interface {
+	GetLayer() uint32
```

Comment:
same here ,layerID, BlockID

Reply:
this causes a bit of conversion hell right now because in the end it all goes to get serialized